src/main/java/org/spongepowered/mod/plugin/SpongePluginContainer.java

```diff
@@ -40,45 +38,52 @@
 import org.spongepowered.api.util.event.Event;
 import org.spongepowered.mod.SpongeMod;
 import org.spongepowered.mod.event.EventRegistry;
-import org.spongepowered.mod.guice.PluginScope;
+import org.spongepowered.mod.guice.SpongePluginGuiceModule;
 
 import java.io.File;
 import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.util.Map;
 
-import javax.inject.Inject;
-
 @NonnullByDefault
 public class SpongePluginContainer extends FMLModContainer implements PluginContainer {
 
     private final Map<String, Object> fmlDescriptor;
-    private final String className;
-    private final ModCandidate container;
     @SuppressWarnings("unused")
     private final File source;
     private Object plugin;
 
-    @Inject
-    private PluginScope scope;
+    private Injector selfInjector;
 
     private LoadController controller;
     @SuppressWarnings("unused")
     private boolean enabled = true;
     private Multimap<Class<? extends Event>, Method> stateEventHandlers = ArrayListMultimap.create();
+    private static final Field LANGUAGE_ADAPTER_FIELD;
+    static {
+        try {
+            LANGUAGE_ADAPTER_FIELD = FMLModContainer.class.getDeclaredField("languageAdapter");
+        } catch (NoSuchFieldException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+        LANGUAGE_ADAPTER_FIELD.setAccessible(true);
+    }
 
     public SpongePluginContainer(String className, ModCandidate container, Map<String, Object> modDescriptor) {
-        // I suggest that you should be instantiating a proxy object, not the real plugin here.
-        super("org.spongepowered.mod.plugin.SpongePluginContainer$ProxyMod", container, modDescriptor);
+        super(className, container, modDescriptor);
```

Comment:
There's more places than just there that check around the class it's given.

We need to either give it a fake class, or provide our own implementations of so much that we might as well stop extending FMLModContainer and implement the whole thing.